packages/rule-no-broken-links/tests/no-broken-links.ts

```diff
@@ -0,0 +1,55 @@
+import { generateHTMLPage } from 'sonarwhal/dist/tests/helpers/misc';
+import { RuleTest } from 'sonarwhal/dist/tests/helpers/rule-test-type';
+import * as ruleRunner from 'sonarwhal/dist/tests/helpers/rule-runner';
+import { getRuleName } from 'sonarwhal/dist/src/lib/utils/rule-helpers';
+
+const ruleName = getRuleName(__dirname);
+
+const bodyWithValidLinks = `<div>
+<a href='https://example.com/'>Example</a>
+</div>`;
+
+const bodyWithImageSource = `<div>
+<img src='https://sonarwhal.com/static/images/next-arrow-c558ba3f13.svg' />
+</div>`;
+
+const bodyWithBrokenLinks = `<div>
+<a href='https://example.com/404'>Example</a>
+</div>`;
+
+const bodyWithBrokenImageSource = `<div>
+<img src='https://example.com/404.png' />
+</div>`;
+
+const bodyWithValidLinksAndBrokenLinks = `<div>
+<a href='https://example.com/'>Example</a>
+<a href='https://example.com/404'>Example2</a>
+</div>`;
+
+const tests: Array<RuleTest> = [
+    {
+        name: `This test should pass`,
```

Comment:
Test names should be more meanful. 

Also we should add some test for [other status codes](https://sonarwhal.com/docs/contributor-guide/how-to/test-rules/#serverconfig).

Reply:
@kshyju You [can configure in the tests the returned status code of a resource](https://sonarwhal.com/docs/contributor-guide/how-to/test-rules/#serverconfig).

e.g.: https://github.com/sonarwhal/sonarwhal/blob/cb524ca3b20a0bb7205561aa813b92b448c9ee8a/packages/rule-no-friendly-error-pages/tests/tests.ts#L84

Reply:
Added more tests to handle other response status codes